woocommerce/woocommerce-android
# Issue/12918 support account creation UI changes

Closes: #12918 

⚠️ Depends on the FluxC changes here: https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/3110


### Description

Adds the functionality to create a new WP.com account from the app when attempting to connect Jetpack and the user enters a non existing WP.com email in the login step. 

Unit tests added in this PR:  https://github.com/woocommerce/woocommerce-android/pull/12945

### Testing information
1. In the app, login into a self hosted site (jurassic ninja site will do) without Jetpack plugin connected
2. Click on Jetpack activation banner
3. Enter an email that is not a WP.com email
4. Check the entered email inbox from the mobile device and click the sign up link
5. Finish the Jetpack installation flow successfully

If you have any doubts check the following screen recording. 

### The tests that have been performed


### Images/gif


https://github.com/user-attachments/assets/e2827d4e-b323-429f-95c7-7fc0d091cfca


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
